File: 0643-maximum-average-subarray-i/0643-maximum-average-subarray-i.ts
function findMaxAverage(nums: number[], k: number): number {

    let sum: number = 0
    let max: number = -Infinity

    for (let i: number = 0; i < nums.length; i++) {

        // add the current value
        sum += nums[i]

        // we have created a sliding window of k size
        if (i+1 >= k) {
            // remove the last trailing value if we are over size k
            if (i >= k) {
                sum -= nums[i - k]
            }
            max = Math.max(sum / k, max)
        }
    }
    return max
};

File: 0643-maximum-average-subarray-i/0643-maximum-average-subarray-i.rb
# @param {Integer[]} nums
# @param {Integer} k
# @return {Float}
def find_max_average(nums, k)
    sum = nums[0..k-1].sum
    max = sum.to_f / k

    (k...nums.length).each do |i|
        sum += nums[i]
            if i >= k
                sum -= nums[i - k]
            end
        max = [max, (sum.to_f/k)].max
        end
    max
end